src/test/scala/com/codacy/parsers/CloverParserTest.scala

```diff
@@ -0,0 +1,65 @@
+package com.codacy.parsers
+
+import java.io.File
+
+import com.codacy.parsers.implementation.CloverParser
+import org.scalatest.{BeforeAndAfterAll, EitherValues, Matchers, WordSpec}
+
+class CloverParserTest extends WordSpec with BeforeAndAfterAll with Matchers with EitherValues {
+  "CloverParser" should {
+    "identify if report is invalid" in {
+      // use some coverage file that does not follow the Clover format
+      val reader = CloverParser.parse(new File("."), new File("src/test/resources/test_cobertura.xml"))
+
+      reader.isLeft shouldBe true
+    }
+
+    "identify if report is valid" in {
+      val reader = CloverParser.parse(new File("."), new File("src/test/resources/test_clover.xml"))
+      reader.isRight shouldBe true
+    }
+
+    "return a valid report" in {
```

Comment:
I would suggest we split this test in multiple individual assertions

Reply:
I split this test into smaller tests. Check if that's what you were thinking.